apps/vaporgui/VolumeEventRouter.cpp

```diff
@@ -21,20 +22,30 @@ VolumeEventRouter::VolumeEventRouter(QWidget *parent, ControlExec *ce)
     
     AddSubtab("Appearance", new PGroup({
         (new PTFEditor)->ShowColormapBasedOnParam(VP::UseColormapVariableTag, false),
-        new PSection("Ray Tracing", {
+        new PSection("Rendering Method", {
             new PStringDropdownHLI<VP>("Raytracing Algorithm", VP::GetAlgorithmNames(VP::Type::DVR), &VP::GetAlgorithm, &VP::SetAlgorithmByUser),
-            new PEnumDropdown(VP::SamplingRateMultiplierTag, {"1x", "2x", "4x", "8x", "16x"}, {1, 2, 4, 8, 16}, "Sampling Rate Multiplier"),
-            (new PDoubleSliderEdit(VP::VolumeDensityTag, "Volume Density"))->EnableDynamicUpdate()->SetTooltip("Changes the overall density or 'opacity' of the volume allowing for finer tuning of the transfer function."),
-            new PCheckbox(VP::UseColormapVariableTag, "Color by other variable"),
-            (new PVariableSelector3D(RenderParams::_colorMapVariableNameTag))->ShowBasedOnParam(VP::UseColormapVariableTag)
         }),
-        (new PColormapTFEditor)->ShowBasedOnParam(VolumeParams::UseColormapVariableTag),
-        new PSection("Lighting", {
-            new PCheckbox(VolumeParams::LightingEnabledTag, "Enabled"),
-            (new PDoubleSliderEdit(VP::PhongAmbientTag,   "Ambient" ))->EnableDynamicUpdate(),
-            (new PDoubleSliderEdit(VP::PhongDiffuseTag,   "Diffuse" ))->EnableDynamicUpdate(),
-            (new PDoubleSliderEdit(VP::PhongSpecularTag,  "Specular"))->EnableDynamicUpdate(),
-            (new PDoubleSliderEdit(VP::PhongShininessTag, "Shininess"))->SetRange(1, 100)->EnableDynamicUpdate()
+        (new PShowIf(VP::_algorithmTag))->Equals(VolumeOSPRay::GetName())->Then({
+            new PSection("OSPRay Parameters", {
+                (new PDoubleSliderEdit(VolumeParams::OSPDensity, "Density"))->SetRange(0, 3)->EnableDynamicUpdate()->SetTooltip("Volume density (aka opacity)."),
+                (new PIntegerSliderEdit("osp_spp", "Samples Per Pixel"))->SetRange(1, 10)->SetTooltip("Number of render passes. Increases fidelity but may significantly reduce performance."),
+                (new PDoubleInput(VolumeParams::OSPSampleRateScalar, "Volume Sample Rate Scalar"))->EnableBasedOnParam("osp_usePT", false)->SetTooltip("Scales the sampling rate along the ray throughout the volume. Increasing may significantly reduce performance."),
```

Comment:
Is there a reason we can't make this into a slider with some reasonable upper limit? In it's present form it is really easy to shoot yourself in the foot by entering an arbitrarily large number.

Reply:
Making it a slider implies that there is a reasonable range and that it makes sense to try changing it within that range. This is a hack since OSPRay sometimes creates an unusable sampling rate. I added heuristics that determine a reasonable default but this allows users to change it so that it is not unusable if my heuristics fail. If I make the slider say 0-1 for some datasets even a value of 0.1 will cause the program to hang for hours whereas for others it may be completely reasonable.

If you insist, I can make it a slider.